src/akushon/action/node/action_node.cpp

```diff
@@ -18,75 +18,61 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-#include <iostream>
+#include "akushon/action/node/action_node.hpp"
+
 #include <iomanip>
+#include <iostream>
 #include <memory>
 #include <string>
 #include <thread>
 #include <vector>
 
-#include "akushon/action/node/action_node.hpp"
-
-#include "akushon/action/node/action_manager.hpp"
 #include "akushon/action/model/action_name.hpp"
 #include "akushon/action/model/pose.hpp"
+#include "akushon/action/node/action_manager.hpp"
 #include "nlohmann/json.hpp"
 #include "rclcpp/rclcpp.hpp"
-#include "tachimawari/joint/model/joint.hpp"
+#include "tachimawari/joint/joint.hpp"
 #include "tachimawari_interfaces/msg/set_joints.hpp"
 
 namespace akushon
 {
 
-std::string ActionNode::get_node_prefix()
-{
-  return "action";
-}
+std::string ActionNode::get_node_prefix() { return "action"; }
 
-std::string ActionNode::run_action_topic()
-{
-  return get_node_prefix() + "/run_action";
-}
+std::string ActionNode::run_action_topic() { return get_node_prefix() + "/run_action"; }
 
-std::string ActionNode::brake_action_topic()
-{
-  return get_node_prefix() + "/brake_action";
-}
+std::string ActionNode::brake_action_topic() { return get_node_prefix() + "/brake_action"; }
 
-std::string ActionNode::status_topic()
-{
-  return get_node_prefix() + "/status";
-}
+std::string ActionNode::status_topic() { return get_node_prefix() + "/status"; }
 
-ActionNode::ActionNode(
-  rclcpp::Node::SharedPtr node, std::shared_ptr<ActionManager> action_manager)
-: node(node), action_manager(action_manager),
-  initial_pose(Pose("initial_pose"))
+ActionNode::ActionNode(rclcpp::Node::SharedPtr node, std::shared_ptr<ActionManager> action_manager)
+: node(node), action_manager(action_manager), initial_pose(Pose("initial_pose"))
 {
-  current_joints_subscriber = node->create_subscription<CurrentJoints>(
-    "/joint/current_joints", 10,
-    [this](const CurrentJoints::SharedPtr message) {
-      {
-        using tachimawari::joint::Joint;
-        std::vector<Joint> current_joints;
-
-        for (const auto & joint : message->joints) {
-          current_joints.push_back(Joint(joint.id, joint.position));
-        }
+  {
+    using tachimawari::joint::JointNode;
 
-        this->initial_pose.set_joints(current_joints);
-      }
-    }
-  );
+    current_joints_subscriber = node->create_subscription<CurrentJoints>(
+      JointNode::current_joints_topic(), 10, [this](const CurrentJoints::SharedPtr message) {
+        {
+          using tachimawari::joint::Joint;
+          std::vector<Joint> current_joints;
+
+          for (const auto & joint : message->joints) {
+            current_joints.push_back(Joint(joint.id, joint.position));
+          }
```

Comment:
It may less readable, but more optimized:
```suggestion
          const auto& joints = message->joints;
          std::vector<Joint> current_joints(joints.size());

          for (size_t i = 0; i < joints.size(); ++i) {
            current_joints[i].id = joints[i].id;
            current_joints[i].position = joints[i].position;
          }
```

Reply:
I have tried it, `Joint` has an explicit constructor and some required parameters are on it. It will cause an error if it is directly initialized with a specific size.